### PR TITLE
Us004 load events

### PR DESCRIPTION
**Pull request by madeye-matt**
@muhamf and my work to implement US004 Load Events.  Quite large

### CODE REVIEW

**Comment by andrewmil** [comment]
Could we update the `Api.md` as part of this PR?

File: src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/impl/CollectionExerciseServiceImpl.java [hunk 1]
```diff
@@ -131,24 +143,54 @@ public CollectionExercise findCollectionExercise(UUID id) {
...
+      if (collex.getScheduledEndDateTime() != null) {
+          collectionExercise.setScheduledEndDateTime(new Timestamp(collex.getScheduledEndDateTime().getTime()));
+      }
+      if (collex.getScheduledExecutionDateTime() != null) {
+          collectionExercise.setScheduledExecutionDateTime(
+                  new Timestamp(collex.getScheduledExecutionDateTime().getTime()));
+      }
+      if (collex.getActualExecutionDateTime() != null) {
+          collectionExercise.setActualExecutionDateTime(new Timestamp(collex.getActualExecutionDateTime().getTime()));
+      }
```
Comment (VibhorMehra) [inline src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/impl/CollectionExerciseServiceImpl.java]: The `actionSvc` fails to retrieve collection exercise dates. Please refer a log file snippet below

`CollectionExerciseClientServiceImpl : cause = null - message = Can not deserialize value of type java.util.Date from String "2018-01-18T17:47:46.976Z": not a valid representation (error: Failed to parse Date value '2018-01-18T17:47:46.976Z': Unparseable date: "2018-01-18T17:47:46.976Z")
 at [Source: {"id":"14fb3e68-4dca-46db-bf49-04b84e07e77c","surveyId":"cb0711c3-0ac8-41d3-ae0e-567e5ea1ef87","name":"BRES_2017","actualExecutionDateTime":"2018-01-18T17:47:46.976Z","scheduledExecutionDateTime":"2017-09-11T00:00:00.000+01:00","scheduledStartDateTime":"2017-09-12T00:00:00.000+01:00","actualPublishDateTime":"2018-01-18T16:24:20.868Z","periodStartDateTime":"2017-09-15T00:00:00.000+01:00","periodEndDateTime":"2017-09-15T23:59:59.000+01:00","scheduledReturnDateTime":"2017-10-06T01:00:00.000+01:00","scheduledEndDateTime":"2018-06-30T00:00:00.000+01:00","executedBy":null,"state":"VALIDATED","caseTypes":[{"actionPlanId":"e71002ac-3575-47eb-b87f-cd9db92bf9a7","sampleUnitType":"B"},{"actionPlanId":"0009e978-0932-463b-a2a1-b45cb3ffcb2a","sampleUnitType":"BI"}],"exerciseRef":"221_201712","userDescription":null,"created":null,"updated":null,"deleted":false}; line: 1, column: 141] (through reference chain: uk.gov.ons.ctp.response.collection.exercise.representation.CollectionExerciseDTO["actualExecutionDateTime"])`

File: pom.xml [hunk 2]
```diff
@@ -24,7 +24,7 @@
     <dependency>
       <groupId>uk.gov.ons.ctp.product</groupId>
       <artifactId>collectionexercisesvc-api</artifactId>
-      <version>10.49.4</version>
+      <version>10.49.5-SNAPSHOT</version>
```
Comment (VibhorMehra) [inline pom.xml]: If we'll be using Events to drive actions, could we please update the `actionSvc` POM as well?

**Comment by VibhorMehra** [comment]
Due to the recent changes delivered on the `collection-instrument` service. The `CollectionInstrumentSvcRestClientImpl` fails when executing collection exercise. Could we please fix this as well? Please refer the log snippet below:

`2018-01-18 12:48:32.720 DEBUG  11630 --- [ask-scheduler-2] c.c.d.DistributedListManagerRedissonImpl : Attempting to relinquish lock on collectionexercisesvc.sample.validation:global:lock
2018-01-18 12:48:32.724 DEBUG  11630 --- [ask-scheduler-2] c.c.d.DistributedListManagerRedissonImpl : Succeeded to relinquish lock on collectionexercisesvc.sample.validation:global:lock
2018-01-18 12:48:32.728 DEBUG  11630 --- [ask-scheduler-2] .g.o.c.r.c.e.c.i.SurveySvcRestClientImpl : about to get to the Survey SVC with surveyId cb0711c3-0ac8-41d3-ae0e-567e5ea1ef87
2018-01-18 12:48:32.734 DEBUG  11630 --- [ask-scheduler-2] .g.o.c.r.c.e.c.i.SurveySvcRestClientImpl : about to get to the Survey SVC with surveyId cb0711c3-0ac8-41d3-ae0e-567e5ea1ef87 and classifierType efa868fb-fb80-44c7-9f33-d6800a17c4da
2018-01-18 12:48:32.780 ERROR  11630 --- [ask-scheduler-2] .i.CollectionInstrumentSvcRestClientImpl : cause = null - message = Unrecognized field "file_name" (class uk.gov.ons.ctp.response.collection.instrument.representation.CollectionInstrumentDTO), not marked as ignorable (3 known properties: "classifiers", "id", "surveyId"])
 at [Source: [
  {
    "classifiers": {
      "COLLECTION_EXERCISE": [
        "14fb3e68-4dca-46db-bf49-04b84e07e77c"
      ], 
      "RU_REF": [
        "49900000002"
      ]
    }, 
    "file_name": null, 
    "id": "7d3433d7-e20f-44a6-82f6-3002e2f3cbcb", 
    "surveyId": "cb0711c3-0ac8-41d3-ae0e-567e5ea1ef87"
  }
]
; line: 14, column: 4] (through reference chain: java.util.ArrayList[0]->uk.gov.ons.ctp.response.collection.instrument.representation.CollectionInstrumentDTO["file_name"])
`

File: src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/impl/CollectionExerciseServiceImpl.java (hunk 1 above)
Comment (madeye-matt) [reply src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/impl/CollectionExerciseServiceImpl.java]: Strictly speaking it is the collection instrument service that is at fault here, as "2018-01-18T17:47:46.976Z" is a valid ISO date.  However I've come to realise that all the services show this same problem - I've fixed this strictness for collection exercise but this has clearly caused issues elsewhere where the date handling is incorrect

**Comment by madeye-matt** [comment]
@VibhorMehra changes made to address these issues - can you retest?

**Comment by VibhorMehra** [comment]
Have tested using `rm-integration` tests. Looks good.

**Pull request merged**